### PR TITLE
DMP-4135: Implement getProfileEntitlements for ARM RPO

### PR DESCRIPTION
**Pull request by chrisbellingham-hmcts**
# [(Dev Only) DMP-4135](https://tools.hmcts.net/jira/browse/DMP-4135)

Implement the ARM RPO getProfileEntitlements service layer.

This PR includes the corresponding Feign client which will be integration tested as part of wider ticket https://tools.hmcts.net/jira/browse/DMP-4143.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jackmaloney — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java
Comment (karen-hedges) [inline src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java:39]: This attribute is included in the ArmApiConfigurationProperties which I have included in my PR for getStorageAccounts, which is just waiting for a second review to get merged.

**Review by karen-hedges — APPROVED** [review]

**Pull request merged**
